src/button/button-contributions.ts

```diff
@@ -169,22 +174,25 @@ export const buttonContributions: ButtonContributionParams[] = [
       id: "new-repo",
       match: /^https?:\/\/([^/]+)\/([^/]+)\/([^/]+)(\/(tree\/.*)?)?$/,
       exampleUrls: [
-        // disabled testing, becuase the new layout doesn't show as an anonymous user
+        // disabled testing, because the new layout doesn't show as an anonymous user
         // "https://github.com/svenefftinge/browser-extension-test",
         // "https://github.com/svenefftinge/browser-extension-test/tree/my-branch",
       ],
-      selector: "#repository-details-container > ul > li:nth-child(6)",
-      containerElement: createElement("div", {
+      selector: "#repository-details-container > ul",
+      containerElement: createElement("li", {
       }),
       application: "github",
       manipulations: [
         {
           // make the code button secondary 
-          element: "#repository-details-container > ul > li:nth-child(5) > get-repo > details > summary",
+          element: "#repository-details-container > ul > li > get-repo > details > summary",
           remove: "Button--primary",
           add: "Button--secondary"
         }
       ],
+      earlyExit: () => {
+        return document.querySelector("div.file-navigation") !== null;
+      }
```

Comment:
I don't like this, it feels like an unnecessary property to maintain. It is a kind of last resort though because the HTML is basically the same and the XPath is stable as well.

Reply:
Ah, I think I get what you are trying here. Could we allow a function as an alternative to a regexp in `match` instead of introducing another property?

Reply:
Sounds sensible. Will merge the properties